src/detect-filemagic.c

```diff
@@ -322,13 +322,20 @@ static void *DetectFilemagicThreadInit(void *data)
 
     (void)ConfGet("magic-file", &filename);
     if (filename != NULL) {
-        SCLogInfo("using magic-file %s", filename);
+        if (strncmp(filename,"",strlen(filename)) == 0) {
```

Comment:
this could be written simpler: strcmp(filename,"") == 0, or just strlen(filename) == 0